File: An2/MPP/mpp-proiect-csharp-DeepSmeag/services/interfaces/IContestsServices.cs
using mpp_proiect_csharp_DeepSmeag.models;
using mpp_proiect_csharp_DeepSmeag.services.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace mpp_proiect_csharp_DeepSmeag.services.interfaces
{
    public interface IContestsServices
    {
        void login(PersoanaOficiu user, IContestsObserver client);

        void logout(PersoanaOficiu user, IContestsObserver client) ;

        List<InscrieriProba> getContests();
        List<Participant> getParticipants();
        List<Participant> getParticipants(int idProba) ;

        void attemptRegisterParticipant(String participantName, int age, InscrieriProba selectedItem);
    }
}

File: An2/MPP/mpp-proiect-java-DeepSmeag/servicesModule/src/main/java/contests/services/IContestsServices.java
package contests.services;

import contests.model.InscrieriProba;
import contests.model.Participant;
import contests.model.PersoanaOficiu;


import java.util.List;

public interface IContestsServices {
    void login(PersoanaOficiu user, String client) throws ContestsException;

    void logout(PersoanaOficiu user, String client) throws ContestsException;

    List<InscrieriProba> getContests() throws ContestsException;
    List<Participant> getParticipants() throws ContestsException;
    List<Participant> getParticipants(int idProba) throws ContestsException;

    void attemptRegisterParticipant(String participantName, int age, List<InscrieriProba> selectedItems) throws ContestsException;
}
